### PR TITLE
GitHub Inline Tips

### PR DESCRIPTION
**Pull request by jdkuki**
Feature request.
Resolves brave/brave-browser#5040

## Submitter Checklist:
- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [x] Linux
  - [ ] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [x] Linux
  - [ ] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [ ] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:
See affected areas in brave/brave-browser#5040

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: browser/extensions/api/brave_rewards_api.cc [hunk 1]
```diff
@@ -158,6 +158,50 @@ ExtensionFunction::ResponseAction BraveRewardsTipRedditUserFunction::Run() {
   return RespondNow(NoArguments());
 }
 
+void BraveRewardsTipRedditUserFunction::OnRedditPublisherInfoSaved(
```
Comment (NejcZdovc) [inline browser/extensions/api/brave_rewards_api.cc]: why is reedit publisher function added?

Comment (NejcZdovc) [reply browser/extensions/api/brave_rewards_api.cc]: oh it was moved?

File: components/brave_rewards/browser/rewards_service_impl.h [hunk 2]
```diff
@@ -651,7 +651,6 @@ class RewardsServiceImpl : public RewardsService,
       SaveMediaInfoCallback callback,
       const ledger::Result result,
       ledger::PublisherInfoPtr publisher);
-
```
Comment (NejcZdovc) [inline components/brave_rewards/browser/rewards_service_impl.h]: please remove

File: browser/extensions/api/brave_rewards_api.cc (hunk 1 above)
Comment (jdkuki) [reply browser/extensions/api/brave_rewards_api.cc]: yeah not sure what happened there

File: components/brave_rewards/resources/extension/brave_rewards/content_scripts/github.ts [hunk 3]
```diff
@@ -0,0 +1,380 @@
...
+  }
+}
+
+const getMemberListItemMetaData = (elem: Element): RewardsTip.MediaMetaData | null => {
+  const ancestor = elem.closest('.table-list-cell')
+  if (ancestor) {
+    const aTags = ancestor.getElementsByTagName('A')
+    if (aTags.length) {
+      const aTag = aTags[0] as HTMLAnchorElement
+      const userName = aTag.href.split('/')[3]
```
Comment (NejcZdovc) [inline components/brave_rewards/resources/extension/brave_rewards/content_scripts/github.ts]: check if href is undefined and if you have array that has 4 items inside

File: components/brave_rewards/resources/extension/brave_rewards/content_scripts/github.ts [hunk 4]
```diff
@@ -0,0 +1,380 @@
...
+  parent.appendChild(tipElem)
+}
+
+const getStarringContainerMetaData = (elem: Element): RewardsTip.MediaMetaData | null => {
+  const ancestor = elem.closest('.d-block')
+  if (ancestor) {
+    const aTags = ancestor.getElementsByTagName('A')
+    if (aTags.length) {
+      const aTag = aTags[0]
+      const userName = (aTag as HTMLAnchorElement).href.split('/')[3]
```
Comment (NejcZdovc) [inline components/brave_rewards/resources/extension/brave_rewards/content_scripts/github.ts]: check if href is undefined and if you have array that has 4 items inside

File: vendor/bat-native-ledger/src/bat/ledger/internal/media/github.cc [hunk 5]
```diff
@@ -377,4 +387,80 @@ void GitHub::SavePublisherInfo(
...
+}
+
+void GitHub::SaveMediaInfo(
+    const std::map<std::string, std::string>& data,
+    ledger::PublisherInfoCallback callback) {
+  auto user_name = data.find("user_name");
+  std::string url = GetProfileAPIURL(user_name->second);
+  ledger_->LoadURL(url,
+                   std::vector<std::string>(),
+                   std::string(),
```
Comment (NejcZdovc) [inline vendor/bat-native-ledger/src/bat/ledger/internal/media/github.cc]: `std::string()` -> `""`

File: vendor/bat-native-ledger/src/bat/ledger/internal/media/github.cc [hunk 6]
```diff
@@ -377,4 +387,80 @@ void GitHub::SavePublisherInfo(
...
+                   std::string(),
```
Comment (NejcZdovc) [inline vendor/bat-native-ledger/src/bat/ledger/internal/media/github.cc]: `std::string()` -> `""`

File: components/brave_rewards/browser/rewards_service_impl.h (hunk 2 above)
Comment (jdkuki) [reply components/brave_rewards/browser/rewards_service_impl.h]: fixed

File: components/brave_rewards/resources/extension/brave_rewards/content_scripts/github.ts (hunk 3 above)
Comment (jdkuki) [reply components/brave_rewards/resources/extension/brave_rewards/content_scripts/github.ts]: fixed

File: components/brave_rewards/resources/extension/brave_rewards/content_scripts/github.ts (hunk 4 above)
Comment (jdkuki) [reply components/brave_rewards/resources/extension/brave_rewards/content_scripts/github.ts]: fixed

File: vendor/bat-native-ledger/src/bat/ledger/internal/media/github.cc (hunk 5 above)
Comment (jdkuki) [reply vendor/bat-native-ledger/src/bat/ledger/internal/media/github.cc]: fixed

File: vendor/bat-native-ledger/src/bat/ledger/internal/media/github.cc (hunk 6 above)
Comment (jdkuki) [reply vendor/bat-native-ledger/src/bat/ledger/internal/media/github.cc]: fixed

**Comment by NejcZdovc** [comment]
@jdkuki please check CI

**Comment by jdkuki** [comment]
> @jdkuki please check CI

fixed lint and rebased with fix for android failure

File: package-lock.json [hunk 7]
```diff
@@ -3579,7 +3579,7 @@
     },
     "@types/filewriter": {
       "version": "0.0.28",
-      "resolved": "https://registry.npmjs.org/@types/filewriter/-/filewriter-0.0.28.tgz",
```
Comment (NejcZdovc) [inline package-lock.json]: Can `package-lock.json` be removes as we didn't changed `package.json`

Comment (jdkuki) [reply package-lock.json]: fixed

**Comment by NejcZdovc** [comment]
@jdkuki can you please address this in a follow up. Don't want to postpone this one any further.

If you do inline tip you get this:

image

Which doesn't make sense as we don't display any data from the github (like we do posts in twitter).

cc @alexwykoff @jenn-rhim for what to do

**Pull request merged**